### PR TITLE
Implement accessing proc properties

### PR DESCRIPTION
**Pull request by ike709**
Not sure if this is the best way to do it but it seems to work.

### CODE REVIEW

File: OpenDreamRuntime/DreamThread.cs [hunk 1]
```diff
@@ -59,6 +59,21 @@ public abstract class DreamProc {
             context.PushProcState(state);
             return context.Resume();
         }
+
+        public DreamValue GetField(string field) {
+            switch (field) {
```
Comment (ike709) [inline OpenDreamRuntime/DreamThread.cs:64]:
```suggestion
            // TODO: Figure out what byond does when these are null
            switch (field) {
```

File: OpenDreamRuntime/DreamThread.cs [hunk 2]
```diff
@@ -61,6 +61,20 @@ public abstract class DreamProc {
...
+                    return new DreamValue(VerbName);
+                case "category":
+                    return new DreamValue(VerbCategory);
+                case "description":
+                    return new DreamValue(VerbDesc);
+                case "invisibility":
+                    return new DreamValue(Invisibility);
+                default:
+                    throw new Exception($"Cannot get field \"{field}\" from {OwningType.ToString()}.{Name}()");
+
```
Comment (SpaceManiac) [inline OpenDreamRuntime/DreamThread.cs:77]: You are missing two closing `}` here

**Comment by ike709** [comment]
I forgot this PR was failing CI.

**Review by SpaceManiac — APPROVED** [review]

**Comment by SpaceManiac** [comment]
possible future notes:
- /tg/ also uses "hidden" since https://github.com/tgstation/tgstation/pull/53112, but I haven't checked whether this genuinely works or not, because
- BYOND returns `null` instead of erroring at runtime if a bad setting is accessed, although it's probably best for OD to error

**Review by wixoaGit — APPROVED** [review]

**Pull request merged**